ledger/participant-integration-api/src/main/scala/platform/indexer/RecoveringIndexer.scala

```diff
@@ -97,9 +95,13 @@ private[indexer] final class RecoveringIndexer(
         }
       } yield ()
 
-    def resubscribeOnFailure(currentSubscription: Resource[IndexFeedHandle]): Unit =
+    def resubscribeOnFailure(
+        currentSubscription: Resource[IndexFeedHandle],
+        actOnSuccess: => Unit,
+    ): Unit =
```

Comment:
I would prefer if this used a separate parameter list for the function and was called with `{ … }` to show that it's a lazily-evaluated function.

```suggestion
    def resubscribeOnFailure(
        currentSubscription: Resource[IndexFeedHandle]
    )(actOnSuccess: => Unit): Unit =
```